modules/swift/pages/vector-search-api-reference.adoc

```diff
@@ -0,0 +1,286 @@
+                        The vectors that do not have the same dimensions 
+                        specified in the config will not be indexed. The
+                        dimensions must be between 2 and 2048 inclusively.
+      @param centroids  The number of centroids which is the number buckets 
+                        to partition the vectors in the index. The number of       
+                        centroids will be based on the expected number of 
+                        vectors to be indexed; one suggested rule is to use 
+                        the square root of the number of vectors. The
+                        centroids must be between 1 and 64000 inclusively. */
+   public init(expression: String, dimensions: UInt, centroids: UInt) { }
+}
+
+----
+
+IMPORTANT: The beta release currently only supports the format of vectors as an array of numbers.
+
+=== Default Value Constants
+
+[source, c]
+----
+
+public class VectorIndexConfiguration {
+  /* Scalar Quantizer */
+  public static let defaultEncoding = VectorEncoding.scalarQuantizer()
+
+  /* Squared Euclidean Distance */
+  public static let defaultMetric = VectorMetric.euclidean   
+}
+
+----
+
+=== Extension Path Configuration
+
+NOTE: The extension configuration is only necessary for the CBL-iOS MacOS and CBL-C platforms as dynamic extension libraries may be at arbitrary locations.
+
+[source, c]
+----
+
+public class CouchbaseLite {
+  /* Registers a directory path to load extension libraries from. Must 
+     be called before using CouchbaseLite but only when using extensions */
+  public static func setExtensionPath(extensionPath: String)
+}
+
+----
+
+== Vector SQL++ Functions
+
+Couchbase Lite currently supports two SQL++ functions, `vector_match()` and `vector_distance()`.
+
+=== Vector Match
+
+==== `vector_match(vectorIndexIdentifier, targetVectorExpr, [limit = 3])`
+
+[cols = "3*"]
+|===
+|Parameter |Is Required |Description
+
+|vectorIndexIdentifier
+|Required
+|The name of the vector index to perform the vector search on.
+|targetVectorExpr
+|Required
+|The target vector expression that returns a vector in the form of an array of numbers.
+|limit
+|Optional
+|The limit number of the returned matched results.
+The maximum number allowed is 10000. An error will be returned when creating a query with a limit greater than 10000.
+
+|===
+
+This function performs vector search against a specific vector index identifier for the specified vector expression.
+If the specified index does not exist, an error will occur on creation of the query.
+The matched vectors will be returned up to the specified limit number, if the limit is not specified then the default value will be used. 
+The returned vectors are sorted by their distance values in ascending order by default.
+
+NOTE: The default value for the `limit` parameter is 3.
+
+[source, sqlpp]
+----
+SELECT name 
+FROM photos 
+WHERE vector_match(photos-index, [0.1, 0.5, 1.2])
+
+----
+
+[source, sqlpp]
+----
+
```

Comment:
These are swift specific but none of examples show any swift code samples.

Reply:
I'm still waiting on these. The api doc only contained C snippets for some reason